### PR TITLE
add redis for realtime/rate limiting

### PR DESCRIPTION
**Pull request by mjdavidson**

### CODE REVIEW

**Comment by consindo** [comment]
Looking good. I would however, take the opportunity to split this out into a new top level project, so we can run with the everything when doing dev locally, as a Docker container, or potentially as a lambda on a cron job.

File: src/waka-realtime/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  redis: Redis
+  prefix: string
+  quotaManager: QuotaManager
+  rateLimiter: <T>(fn: () => Promise<T>) => Promise<T>
+  region: BaseRealtime
+  logger: Logger
+  constructor(config: RealtimeConfig) {
+    const logger = createLogger(config.prefix, config.version)
+    this.logger = logger
+    this.prefix = config.prefix
```
Comment (consindo) [inline src/waka-realtime/index.ts:41]: I would probably also prepend a waka-rt- to the key too? Just in case we need to use it in other components.

**Comment by mjdavidson** [comment]
you can have a look over this and merge when u want

**Review by consindo — CHANGES_REQUESTED** [review]
Functionality is excellent - well done! Well architected too, and just works with the orchestrator.

Only a couple of things:
- [x] Add a section to the readme about waka-realtime
- [x] I think `waka-worker/realtime/ProtobufRealtime.ts` needs to be deleted
- [x] If the Redis connection fails, it shouldn't crash the process, but rather just not do realtime.
- [x] As the Redis connection is made in the constructor, it makes it harder to unit test. I would move the connection to the `start()` method, and close it in the `stop()` method.
- [x] You'll need to add redis configuration support to the configManager or something, otherwise we won't be able to use elasticache. It would be good if you could implement in a similar way to the database connections (i.e pointers to configs)

**Review by consindo — APPROVED** [review]

**Pull request merged**